src/urllib3/util/util.py

```diff
@@ -25,7 +26,7 @@ def to_str(
     return x.decode()
 
 
-def reraise(tp, value, tb=None):
+def reraise(tp: type, value: Exception, tb: Optional[TracebackType] = None) -> NoReturn:
```

Comment:
This function comes from six, and [typeshed uses the following types](https://github.com/python/typeshed/blob/0b1cd5989669544866213807afa833a88f649ee7/stubs/six/six/__init__.pyi#L93-L95):


```python3
def reraise(
    tp: Optional[Type[BaseException]], value: Optional[BaseException], tb: Optional[types.TracebackType] = ...
) -> NoReturn: ...
```

Maybe we should just copy that?

Reply:
Thanks @pquentin for the suggestion. I will change it